tools/integration-tests/tester/tests/value/value_test.go

```diff
@@ -0,0 +1,121 @@
+package autopeering
+
+import (
+	"testing"
+	"time"
+
+	"github.com/iotaledger/goshimmer/tools/integration-tests/tester/tests"
+	"github.com/stretchr/testify/require"
+)
+
+// TestValueIotaPersistence issues messages on random peers, restarts them and checks for persistence after restart.
+func TestValueIotaPersistence(t *testing.T) {
+	n, err := f.CreateNetwork("valueIota_TestPersistence", 4, 2)
+	require.NoError(t, err)
+	defer tests.ShutdownNetwork(t, n)
+
+	// wait for peers to change their state to synchronized
+	time.Sleep(5 * time.Second)
+
+	// master node sends funds to all peers in the network
+	txIds, addrBalance := tests.SendValueMessagesOnFaucet(t, n.Peers())
```

Comment:
I think here it shows itself that the static transfer value is hidden from the user of this function. It makes more sense to define the amount to be sent around as otherwise this function could be called while the senders do not actually have the minimum amount. This then causes the function to hang indefinitely because of the "if fail {i--}" condition in the send functions.